src/module.cpp

```diff
@@ -41,12 +41,83 @@
 #include <set>
 #include <utility>
 #include <unordered_set>
+#include <bitset>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_TRACE_FINALIZE)
 
+template <std::size_t N>
+struct bit_signal
```

Comment:
Unit tests for this struct would be good

Reply:
Tests added.